engine/debug/debug_party.asm

```diff
@@ -21,4 +21,139 @@ DebugTeam:
 	db -1 ; end
 
 DebugStart:
+IF DEF(_DEBUG)
+	xor a ; PLAYER_PARTY_DATA
+	ld [wMonDataLocation], a
+
+	; Fly anywhere.
+	dec a ; $ff
+	ld [wTownVisitedFlag], a
+	ld [wTownVisitedFlag + 1], a
+
+	; Get all badges except Earth Badge.
+	ld a, $ff ^ (1 << BIT_EARTHBADGE)
+	ld [wObtainedBadges], a
+
+	call SetDebugTeam
+
+	; Pikachu gets Surf.
+	ld a, SURF
+	ld hl, wPartyMon4Moves + 2
+	ld [hl], a
+
+	; Snorlax gets four HM moves.
+	ld hl, wPartyMon1Moves
+	ld a, FLY
+	ld [hli], a
+	ld a, CUT
+	ld [hli], a
+	ld a, SURF
+	ld [hli], a
+	ld a, STRENGTH
+	ld [hl], a
+
+	; Get some debug items.
+	ld hl, wNumBagItems
+	ld de, DebugItemsList
+.items_loop
+	ld a, [de]
+	cp -1
+	jr z, .items_end
+	ld [wcf91], a
+	inc de
+	ld a, [de]
+	inc de
+	ld [wItemQuantity], a
+	call AddItemToInventory
+	jr .items_loop
+.items_end
+
+	; Complete the Pokédex.
+	ld hl, wPokedexOwned
+	call DebugSetPokedexEntries
+	ld hl, wPokedexSeen
+	call DebugSetPokedexEntries
+	SetEvent EVENT_GOT_POKEDEX
+
+	; Player chose Pikachu.
+	ld hl, wRivalStarter
+	ld a, 1
+	ld [hli], a
+	ld a, NUM_POKEMON
+	ld [hli], a ; ???
```

Comment:
Any guesses what the purpose of this might be?

Reply:
I don't know, the byte after `wRivalStarter` is otherwise unused. How about giving it a label in debug, like:

```
wRivalStarter::
	ds 1

IF DEF(_DEBUG)
; This byte gets set to NUM_POKEMON by DebugStart.
wUnknownDebugByte::
ENDC
	ds 1
```

and likewise comment `ld [hli], a ; sets wUnknownDebugByte`.